uaclient/cli.py

```diff
@@ -824,6 +833,14 @@ def action_config_set(args, *, cfg, **kwargs):
                 ", ".join(config.UA_CONFIGURABLE_KEYS)
             )
         )
+    apt_proxies_config_options = (
+        "apt_http_proxy",
+        "apt_https_proxy",
+        "ua_apt_http_proxy",
+        "ua_apt_https_proxy",
+        "global_apt_http_proxy",
+        "global_apt_https_proxy",
+    )
```

Comment:
I think the logic might be easier if we don't treat this as one big group. With everything mixed together it is hard to read
I suggest that we instead treat them as separate groups, something like this:

```python
ua_scoped_apt_proxy_options = ("ua_apt_http_proxy", "ua_apt_https_proxy")
global_scoped_apt_proxy_options = ("global_apt_http_proxy", "global_apt_https_proxy")
deprecated_global_scoped_apt_proxy_options = ("apt_http_proxy", "apt_https_proxy")

...


elif set_key in ua_scoped_apt_proxy_options:
    # ua scoped logic
elif (
    set_key in global_scoped_apt_proxy_options or
    set_key in deprecated_global_scoped_apt_proxy_options
):
    if set_key in deprecated_global_scoped_apt_proxy_options:
        # apt_* warnings
        set_key = "global_" + set_key
    # global scoped logic

```